internal/workload/manager.go

```diff
@@ -77,49 +77,58 @@ func (w *WorkloadManager) GetExportedHostPath(workloadName string) string {
 	return volumes.HostPathVolumePath(w.volumesDir, workloadName)
 }
 
-func (w *WorkloadManager) Update(configuration models.DeviceConfigurationMessage) error {
+func (w *WorkloadManager) Update(configuration models.DeviceConfigurationMessage) []error {
 	w.managementLock.Lock()
 	defer w.managementLock.Unlock()
-
+	errors := []error{}
 	configuredWorkloadNameSet := make(map[string]struct{})
 	for _, workload := range configuration.Workloads {
 		log.Tracef("Deploying workload: %s", workload.Name)
 		configuredWorkloadNameSet[workload.Name] = struct{}{}
-		// TODO: change error handling from fail fast to best effort (deploy as many workloads as possible)
+
 		pod, err := w.toPod(workload)
 		if err != nil {
-			return err
+			errors = append(errors, fmt.Errorf(
+				"Cannot convert workload '%s' to pod, err: %v", workload.Name, err))
```

Comment:
same comment as earlier about capitalization and discarding the `, err` substring